README.md
# PersonalWebsite
Watch my website change while I make small modifications...


public/changeLog.txt
Change Log:
-changed May 2016 on top of the signature to be June 2016
-changed text content a little bit
-added Facebook link to the social media links

Change Log:
-fixed the google maps tint showing non-tinted spaces at the top and the bottom on a mobile device
-modal images now show the correct image instead of the QNote logo

Change Log:
-clicking the social media links now open them in a new tab
-modalInjector now injects correct values for QNote
-added padding between Skills divs
-changed website text content. getting ready for deployment

Change Log:
-added tint to contact background div
-added signature to the footer of the page
-initial steps on modalInjector.js. I did not want duplicate HTML code for all the modals for the projects so this should be a better solution

Change Log:
-Contact area now has a static Google Maps background
-Fixed IP Manager's modal, added modals for all ongoing projects. However, decided this is not a good approach. Started writing modalInjector.js

Change Log:
-removed social media buttons under Google Maps profile picture
-each social shows its own logo colour on mouse hover
-rocky journey logo layout has been fixed

Change Log:
-Rocky journey app icon has been added. Will fix the layout soon
-Resume link now works

Change Log:
-Fixed the Resume button displaying wrong colours and underline on mouse hover
-Social media icons are not wrapping to the next line when width < 960
-Changed projects section's background colour

Change Log:
-changed header background image to the "wall of self-motivated creativity"
-changed navbar colour to something nicer
-added github logo and fixed the social media links except facebook
-adjusted the resume button colour and size

public/js/modalInjector.js
var modalTemplate = '<div class=\"modal fade\" id=\"#modal-id\" tabindex=\"-1\" role=\"dialog\" aria-labelledby=\"myModalLabel\" aria-hidden=\"true\"><div class=\"modal-dialog\"><div class=\"modal-content\"><div class=\"modal-header\"><button type=\"button\" class=\"close\" data-dismiss=\"modal\" aria-hidden=\"true\"></button> <h4 class=\"modal-title\">#modal-title</h4></div><div class=\"modal-body row\"><img style=\"margin-top: 70px;\" src=\"images/#image-filename\" name=\"aboutme\" class=\"img-responsive col-md-4\"/><h3 class=\"media-heading\">#app-name</h3><h4 class=\"text-muted\">#header</h4><p>#description</p><hr><span><strong>#list-1-text</strong></span>#technologies-used<br><span><strong>#list-2-text</strong></span>#download-for<hr><span><strong>#list-3-text</strong></span><br>#image-links</div><div class=\"modal-footer\"><center><button type=\"button\" class=\"btn btn-default\" data-dismiss=\"modal\">Close</button></center></div></div></div></div>';

var imageTemplate = '<img src=\"#link\"height=\"120\"/>';
var labelTemplate = '<span class=\"label label-#type\">#inner-text</span>';

console.log('Injection Started');

document.onload = init();

function injectModal(newModal) {
    $(newModal).appendTo('body');
}

function createModal(modalId, appName, category, description, logoLocation, slogan, screenshots, list1Content, list2Content, list1Text, list2Text, list3Text) {
    var newModal = modalTemplate;
    newModal = newModal.replace('#modal-id', modalId);
    newModal = newModal.replace('#modal-title', category);
    newModal = newModal.replace('#image-filename', logoLocation);
    newModal = newModal.replace('#description', description);
    newModal = newModal.replace('#app-name', appName);
    newModal = newModal.replace('#header', slogan);

    var imageLinks = '';

    for (var i in screenshots) {
        imageLinks += (imageLinks, this.imageTemplate.replace('#link', screenshots[i]));
    }

    newModal = newModal.replace('#image-links', imageLinks);

    var techUsed = '';
    var downFor = '';

    for (var i = 0; i < list1Content.length; i++) {
        techUsed += (techUsed, this.labelTemplate.replace('#inner-text', list1Content[i][0]).replace('#type', list1Content[i][1]))
    }

    for (var i = 0; i < list2Content.length; i++) {
        downFor += (downFor, this.labelTemplate.replace('#inner-text', list2Content[i][0]).replace('#type', list2Content[i][1]))
    }

    newModal = newModal.replace('#technologies-used', techUsed);
    newModal = newModal.replace("#download-for", downFor);

    newModal = newModal.replace('#list-1-text', list1Text);
    newModal = newModal.replace('#list-2-text', list2Text);
    newModal = newModal.replace('#list-3-text', list3Text);

    injectModal(newModal);
}

//requests the .json object from the server
function loadJSON(callback) {

    var xobj = new XMLHttpRequest();
    xobj.overrideMimeType('application/json');
    xobj.open('GET', 'js/modals.json', true); // Replace 'my_data' with the path to your file
    xobj.onreadystatechange = function () {
        if (xobj.readyState == 4 && xobj.status == '200') {
            // Required use of an anonymous callback as .open will NOT return a value but simply returns undefined in asynchronous mode
            callback(xobj.responseText);
        }
    };
    xobj.send(null);
}

function init() {
    loadJSON(function (response) {
        // Parse JSON string into object
        var actual_JSON = JSON.parse(response);

        for (var i = 0; i < actual_JSON.length; i++) {
            var current = actual_JSON[i];
            createModal(current.modalId, current.title, current.category, current.description, current.logoLocation, current.slogan, current.screenshots, current.list_1_content, current.list_2_content, current.list_1_text, current.list_2_text, current.list_3_text);
        }
    });
}

public/requiredFiles/BetterCaseWithRegex_.js
document.getElementById("doIT").onclick = function() {updateView()};

var resultP = document.getElementById("result");
var inputForm = document.getElementById("inputForm");
var alphabetArr = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '@'];

//algorithm with logn complexity to check if the number is a prime.
//proof and pseudocode can be found here: https://en.wikipedia.org/wiki/AKS_primality_test
function isPrimeAKG(n)
{
    if (n <= 3)
        return n > 1;
    else if (n % 2 == 0 || n % 3 == 0)
        return false;
    else
    {
        for (var i = 5; i * i < n; i += 6) {
            if (n % i == 0 || n % (i + 2) == 0)
                return false;
        }
        return true;
    }
}

//javascript is dynamically typed. can return both integers and boolean valeus in the same function
function checkInputValidityRegexAdvanced(input)
{
    for (var i = 0; i < input.length; i++) {
        if (alphabetArr.indexOf(input[i]) == -1) {
            return false;
        }
    }
    return true;
}

//remember to check for the overflow!!! - we're allowed to enter 10 integers at most. no need for check overhead.
function getSum(input)
{
    var sum = 0;
    if (checkInputValidityRegexAdvanced(input)) {
        for (var i = 0; i < input.length; i++)
            if(input.charAt(i) == '@')
                sum += 50;
            else
                sum += input.charCodeAt(i) - 96;
    }
    return sum;
}

function getResult(input)
{
    var validityResult = checkInputValidityRegexAdvanced(input);
    if (validityResult == true)
    {
        var sum = getSum(input);
        return "the sum is: " + sum + " and it " + (isPrimeAKG(sum) ? "is " : "is not ") + "a prime";
    }
    else if (validityResult == false)
        return "only enter letters from the English alphabet [a-z]";
    else if (validityResult == -1)
        return "you can enter a maximum of 10 characters";
    else if (validityResult == -2)
        return "the text field is empty";
    else
        return "unknown error. what might this be?";

}

//when the user presses enter key or clicks the calculcate button...
function updateView()
{
    resultP.innerHTML = getResult(inputForm.value);
}

public/js/scrolling-nav.js
$(window).scroll(function() {
    if ($(".navbar").offset().top > 50) {
        $(".navbar-fixed-top").addClass("top-nav-collapse");
    } else {
        $(".navbar-fixed-top").removeClass("top-nav-collapse");
    }
});

$(function() {
    $('a.page-scroll').bind('click', function(event) {
        var $anchor = $(this);
        $('html, body').stop().animate({
            scrollTop: $($anchor.attr('href')).offset().top
        }, 1500, 'easeInOutExpo');
        event.preventDefault();
    });
});

var modalTemplate = '<div class=\"modal fade\" id=\"#modal-id\" tabindex=\"-1\" role=\"dialog\" aria-labelledby=\"myModalLabel\" aria-hidden=\"true\"><div class=\"modal-dialog\"><div class=\"modal-content\"><div class=\"modal-header\"><button type=\"button\" class=\"close\" data-dismiss=\"modal\" aria-hidden=\"true\"></button> <h4 class=\"modal-title\">#modal-title</h4></div><div class=\"modal-body row\"><img style=\"margin-top: 70px;\" src=\"images/#image-filename\" name=\"aboutme\" class=\"img-responsive col-md-4\"/><h3 class=\"media-heading\">#app-name</h3><h4 class=\"text-muted\">#header</h4><p>#description</p><hr><span><strong>#list-1-text</strong></span>#technologies-used<br><span><strong>#list-2-text</strong></span>#download-for<hr><span><strong>#list-3-text</strong></span><br>#image-links</div><div class=\"modal-footer\"><center><button type=\"button\" class=\"btn btn-default\" data-dismiss=\"modal\">Close</button></center></div></div></div></div>';

var imageTemplate = '<img src=\"#link\"height=\"120\"/>';
var labelTemplate = '<span class=\"label label-#type\">#inner-text</span>';

console.log('Injection Started');

init();

function injectModal(newModal) {
    document.body.innerHTML += newModal;
}

function createModal(modalId, appName, category, description, logoLocation, slogan, screenshots, list1Content, list2Content, list1Text, list2Text, list3Text) {
    var newModal = modalTemplate;
    newModal = newModal.replace('#modal-id', modalId);
    newModal = newModal.replace('#modal-title', category);
    newModal = newModal.replace('#modal-title', category);
    newModal = newModal.replace('#image-filename', logoLocation);
    newModal = newModal.replace('#description', description);
    newModal = newModal.replace('#app-name', appName);
    newModal = newModal.replace('#header', slogan);

    var imageLinks = '';

    for (var i in screenshots) {
        imageLinks += (imageLinks, this.imageTemplate.replace('#link', screenshots[i]));
    }

    newModal = newModal.replace('#image-links', imageLinks);

    var techUsed = '';
    var downFor = '';

    for (var i = 0; i < list1Content.length; i++) {
        techUsed += (techUsed, this.labelTemplate.replace('#inner-text', list1Content[i][0]).replace('#type', list1Content[i][1]))
    }

    for (var i = 0; i < list2Content.length; i++) {
        downFor += (downFor, this.labelTemplate.replace('#inner-text', list2Content[i][0]).replace('#type', list2Content[i][1]))
    }

    newModal = newModal.replace('#technologies-used', techUsed);
    newModal = newModal.replace("#download-for", downFor);

    newModal = newModal.replace('#list-1-text', list1Text);
    newModal = newModal.replace('#list-2-text', list2Text);
    newModal = newModal.replace('#list-3-text', list3Text);

    injectModal(newModal);
}

//requests the .json object from the server
function loadJSON(callback) {

    var xobj = new XMLHttpRequest();
    xobj.overrideMimeType('application/json');
    xobj.open('GET', 'js/modals.json', true); // Replace 'my_data' with the path to your file
    xobj.onreadystatechange = function () {
        if (xobj.readyState == 4 && xobj.status == '200') {
            // Required use of an anonymous callback as .open will NOT return a value but simply returns undefined in asynchronous mode
            callback(xobj.responseText);
        }
    };
    xobj.send(null);
}

function init() {
    loadJSON(function (response) {
        // Parse JSON string into object
        var actual_JSON = JSON.parse(response);

        for (var i = 0; i < actual_JSON.length; i++) {
            var current = actual_JSON[i];
            createModal(current.modalId, current.title, current.category, current.description, current.logoLocation, current.slogan, current.screenshots, current.list_1_content, current.list_2_content, current.list_1_text, current.list_2_text, current.list_3_text);
        }
    });
}